firmware/application/ui_navigation.cpp

```diff
@@ -555,6 +560,76 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"SYS";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        (sd_status == sd_card::Status::Mounted)) {
+        nav_.display_modal(
+            "Warning",
+            "        WARNING!!!\n"
+            "Sdcard content changed\n"
+            "in this update,\n"
+            "Press YES to auto move.\n"
+            "After finished,\n"
+            "you should move your own fil-\n"
+            "es back into original folder,\n"
+            "which is now only for users.\n"
+            "You can also choose NO,\n"
+            "and then manually merge\n"
+            "following the Wiki.\n"
+            "Details:\n"
+            "Mayhem wiki - Updating",
+            COMPACTYESNO,
+            [this](bool choice) {
+                if (choice) {
+                    new_sdcard_structure_worker();
+                }
+            });
+    }
+}
+
+void SystemStatusView::new_sdcard_structure_worker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path system_dir = u"SYS";
+
+    // TODO: Use blacklist or whitelist?
+    std::vector<std::filesystem::path> ignore_dirs = {u"FIRMWARE",
+                                                      u"APPS",
+                                                      u"SYS",
+                                                      u"SETTINGS",
+                                                      u"DEBUG",
+                                                      u"LOGS",
+                                                      u"CAPTURES",
+                                                      u".Trash-1000",
+                                                      u"SCREENSHOTS"};
+
+    /// worker moving folders
+    ensure_directory(system_dir);
+
+    auto directories = scan_root_directories(root_dir);
+    for (const auto& dir_entry : directories) {
+        if (std::find(ignore_dirs.begin(), ignore_dirs.end(), dir_entry) == ignore_dirs.end()) {
+            std::filesystem::path new_path = system_dir / dir_entry.filename();
+            rename_file(dir_entry, new_path);
+        }
+    }
+
+    /// worker renaming
+    rename_file(u"/SYS/SAMPLES", u"/SYS/CAPTURES");  // it's because, make it easier to use the sample folder.
```

Comment:
Not sure about this one, but I think I preferred having the SAMPLES in the SAMPLES folder.

Reply:
This folder used to be SAMPLES, with our sample C16 files.
The reason I rename it, is because it would cost too much memory to hard coded the dir mapping (for Profile switcher func), in which means, the Profile switch func is automatically, without any hard coded dir mapping

Mapping: for example
/SYS/SAMPLES <---> /CAPTURES

if they don't have same name with different level, it will need hard coded, which is not worth